openedx/core/djangoapps/user_authn/views/tests/test_utils.py

```diff
@@ -0,0 +1,44 @@
+"""
+Tests for user utils functionality.
+"""
+from django.test import TestCase
+from datetime import datetime
+from unittest.mock import patch
+from openedx.core.djangoapps.user_authn.views.utils import generate_username_from_data
+import ddt
+
+
+@ddt.ddt
+class TestGenerateUsername(TestCase):
+    """
+    Test case for the generate_username_from_data function.
+    """
+
+    @patch('openedx.core.djangoapps.user_authn.views.utils.generate_username')
+    def test_generate_username(self, mock_generate_username):
+        """
+        Test generate_username_from_data function with mock generate_username.
+        """
+
+        mock_generate_username.return_value = 'JD_202403_XyZa'
+        # Test with known inputs
+        data = {'first_name': 'John', 'last_name': 'Doe'}
+        username = generate_username_from_data(data)
+
+        self.assertEqual(username, 'JD_202403_XyZa')
+
+    @ddt.data(
+        ({'first_name': 'John', 'last_name': 'Doe'}, "JD"),
+        ({'name': 'Jane Smith'}, "JS"),
+        ({'name': 'Jane'}, "J"),
+        ({'name': 'John Doe Smith'}, "JD")
+    )
+    @ddt.unpack
+    def test_generate_username_from_data(self, data, expected_initials):
```

Comment:
Can we add a unit test to test exception if the username is already exist?

Reply:
@mubbsharanwar: These tests have already been implemented. My current task is focused on generating the username. Once the username is generated, the subsequent flow remains consistent. Upon passing all data to the `create_account` method, it is expected to raise an exception.